### ISSUE TITLE
strange work with .pyi files

### ISSUE THREAD

**Issue by ZeN220** [python/mypy#16672]
I generate stub files for GRPC client/server with [mypy-protobuf](https://github.com/nipunn1313/mypy-protobuf). As far as I can tell, they are generated correctly, and mypy should detect the Response and Data classes, for they are defined in the .pyi file

**To Reproduce**
https://gist.github.com/ZeN220/5ed2f6b4f841929860ce111d16940aaa

**Actual Behavior**
```
> mypy a_pb2.py
__init__.py:1: error: Module "aaa.a_pb2" has no attribute "Response"  [attr-defined]
__init__.py:1: error: Module "aaa.a_pb2" has no attribute "Data"  [attr-defined]
Found 2 errors in 1 file (checked 1 source file)
```
But if running:
```
> mypy .
Success: no issues found in 2 source files
```

- Mypy version used: 1.7.1
- Mypy command-line flags: nothing
- Mypy configuration options from `mypy.ini` (and other config files): 
```toml
[tool.mypy]
disable_error_code = ["assignment"]
```
- Python version used: 3.9.13

**Comment by erictraut** [comment]
It looks like you're trying to run mypy on the original `a_pb2.py` file. The `a_pb2.pyi` stub is a stand-in for this file and is used when you type check _other files_ that import `a_pb2.py`. This stub cannot be used to type check `a_pb2.py` itself. In fact, you shouldn't try to run a type checker on `a_pb2.py`, which contains generated (untyped) code.

**Comment by ZeN220** [comment]
> It looks like you're trying to run mypy on the original `a_pb2.py` file. The `a_pb2.pyi` stub is a stand-in for this file and is used when you type check _other files_ that import `a_pb2.py`. This stub cannot be used to type check `a_pb2.py` itself. In fact, you shouldn't try to run a type checker on `a_pb2.py`, which contains generated (untyped) code.

hm, I am using pre-commit and it automatically substitutes the names of all changed files. does this mean I should exclude all files that have .pyi counterparts?

**Comment by erictraut** [comment]
> does this mean I should exclude all files that have .pyi counterparts?

You should exclude any files that are not meant to be type checked. That includes any generated `*_pb2.py` files.

**Comment by ZeN220** [comment]
thank you for your prompt help!
